### ISSUE TITLE
Indexing by 1-dimension `int64` tensor returns incorrect shape

### ISSUE THREAD

**Issue by episodeyang** [pytorch/pytorch#20697]
## 🐛 Bug

Sometimes indexing by a 1-dimension `int64` tensor with a single element returns a tensor with reduced rank.

## To Reproduce

<img width="498" alt="Screenshot 2019-05-19 21 41 52" src="https://user-images.githubusercontent.com/630490/57996908-ec462a00-7a7e-11e9-8ba2-1a7bf5f71bca.png">

This bug gets even better:

<img width="228" alt="Screenshot 2019-05-19 21 44 09" src="https://user-images.githubusercontent.com/630490/57996986-4515c280-7a7f-11e9-962d-75f60589a4a2.png">

```python
obs.dtype
Out[28]: dtype('float32')
obs.shape
Out[29]: (54, 1, 64, 64)
a.shape
Out[30]: torch.Size([2])
obs.shape
Out[31]: (54, 1, 64, 64)
b = torch.tensor([3])
obs[b].shape
Out[33]: (1, 64, 64)
a = torch.tensor([3, 3])
obs[a].shape
Out[35]: (2, 1, 64, 64)
```

## Environment

Collecting environment information...
PyTorch version: 1.1.0
Is debug build: No
CUDA used to build PyTorch: 10.0.130

OS: Ubuntu 18.04.1 LTS
GCC version: (Ubuntu 7.4.0-1ubuntu1~18.04) 7.4.0
CMake version: version 3.10.2

Python version: 3.7
Is CUDA available: Yes
CUDA runtime version: 10.0.130
GPU models and configuration:
GPU 0: Quadro GP100
GPU 1: Quadro GP100

Nvidia driver version: 410.79
cuDNN version: Could not collect

Versions of relevant libraries:
[pip] numpy==1.16.2
[pip] torch==1.1.0
[pip] torchvision==0.2.2
[conda] mkl                       2019.3                      199
[conda] pytorch                   1.1.0           py3.7_cuda10.0.130_cudnn7.5.1_0    pytorch
[conda] pytorch-nightly           1.1.0.dev20190411 py3.7_cuda10.0.130_cudnn7.4.2_0    pytorch
[conda] torchvision               0.2.2                      py_3    pytorch

**Comment by colesbury** [comment]
This looks like the expected behavior?

**Comment by colesbury** [comment]
Also please update the "steps to reproduce" so that it's actually reproducible. For example, don't use images of code and include the full definition of variables (objs, a, etc.). The "steps to reproduce" should preferably be a self-contained script that someone else can run. It also helps to point out where the behavior is different than expected.

**Comment by gchanan** [comment]
I couldn't reproduce this, i.e.

```
In [31]: torch.zeros((54, 1, 64, 64))[torch.tensor([3])].shape
Out[31]: torch.Size([1, 1, 64, 64])

In [33]: np.zeros((54, 1, 64, 64))[np.array([3])].shape
Out[33]: (1, 1, 64, 64)
```

so I'm going to close, please reopen if I've done something incorrect.

**Comment by episodeyang** [comment]
I couldn’t reproduce this on a Mac either. This only occurs during run time
on learnfair, with CUDA loaded.

The problem is persistent and the screenshot shows the problem does exist.

**Comment by colesbury** [comment]
@episodeyang what exactly is the problem?

**Comment by episodeyang** [comment]
Hey Sam, I tried pretty hard to reproduce this, but it seems that it only happens with an actual run. This is why I attached the screenshot to show the problem.

The issue is that somehow, if you have a 1-dimensional tensor with only 1 element, and you use it to index another tensor, the returned tensor is short on one dimension. But casting the tensor to list works:

![](https://user-images.githubusercontent.com/630490/57996986-4515c280-7a7f-11e9-962d-75f60589a4a2.png)

The shape of the tensor looks like below:

![](https://user-images.githubusercontent.com/630490/57996908-ec462a00-7a7e-11e9-8ba2-1a7bf5f71bca.png)

So far I tried to reproduce this issue 

1. on my mac 
2. in an iPython console on `learnfair` with `cuda` enabled.

But neither demonstrates the reported issue.
